pkg/controller/registry/resolver/instrumented_resolver_test.go

```diff
@@ -0,0 +1,55 @@
+package resolver
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/stretchr/testify/require"
+)
+
+var result float64
```

Comment:
It looks like this is a global variable because the NewInstrumentedResolver needs be passed a few functions that can't take the `result` variable. If you want to avoid a global variable, you could create the functions like this: https://play.golang.org/p/bfMdb_P3Vcx

Reply:
@awgreene note that the functions `changeToFailure` and `changeToSuccess` have the same signature as `RegsiterDependencyResolutionSucceeded` and `RegisterDependencyResolutionFalied`. The signature for those methods need to be changed too to accommodate this design. 

I'll leave this quote as my supporting argument for not doing that: https://stackoverflow.com/a/2325845

Reply:
You might consider moving some of these identifiers inside the test function body so that two tests are not mutating and reading the same state.

Reply:
> You might consider moving some of these identifiers inside the test function body so that two tests are not mutating and reading the same state.

This is my main concern. No matter how clear you try to make things, it can become significantly less clear when multiple tests can manipulate the state of a common variable.

Reply:
> The signature for those methods need to be changed too to accommodate this design.

The signatures aren't changed?